patches/minecraft/net/minecraft/inventory/container/CraftingResultSlot.java.patch

```diff
@@ -1,10 +1,14 @@
 --- a/net/minecraft/inventory/container/CraftingResultSlot.java
 +++ b/net/minecraft/inventory/container/CraftingResultSlot.java
-@@ -43,6 +_,7 @@
+@@ -43,6 +_,11 @@
     protected void func_75208_c(ItemStack p_75208_1_) {
        if (this.field_75237_g > 0) {
           p_75208_1_.func_77980_a(this.field_75238_b.field_70170_p, this.field_75238_b, this.field_75237_g);
-+         net.minecraftforge.fml.hooks.BasicEventHooks.firePlayerCraftingEvent(this.field_75238_b, p_75208_1_, this.field_75239_a);
++         net.minecraft.item.crafting.IRecipe<?> recipe = null;
++         if (this.field_75224_c instanceof IRecipeHolder) {
++            recipe = ((IRecipeHolder)this.field_75224_c).func_193055_i();
++         }
++         net.minecraftforge.fml.hooks.BasicEventHooks.firePlayerCraftingEvent(this.field_75238_b, p_75208_1_, this.field_75239_a, recipe);
```

Comment:
Couldn't this patch be mitigated by moving the check if it is a recipe holder into the event hook and just adding the inventory container itself as a parameter?

Reply:
Is BasicEventHooks intended to be used by modders or just internal infrastructure to move code out of the patches?
If it's not intended to be used by modders, moving the cast there would make sense.

Reply:
Alright. Done. The patch is now a one-liner again. I went with the assumption that BasicEventHooks is not for modders to use.